### PR TITLE
I5772-licenses that arent links are breaking pages

### LINKED ISSUE #5772: Addons that have the All Rights Reserved license are now showing up broken
### Describe the problem and steps to reproduce it:

Addons that have the "All Rights Reserved" license are showing up broken 

eg:
https://addons-dev.allizom.org/en-US/firefox/addon/wide-zmeurica/
https://addons-dev.allizom.org/en-US/firefox/addon/source/
https://addons-dev.allizom.org/en-US/firefox/addon/rm-theme-new-sizes/

### What happened?

I think this is related to the recent Router changes and Link being a little bit more strict.

### What did you expect to happen?

Pages would appear

### PR DESCRIPTION
**Pull request by rebmullin**
fixes #5772

## BEFORE - extension
Alt text

## BEFORE - static theme
Alt text

## AFTER - extension
Alt text

## AFTER - static theme
Alt text

originally I updated just the AddonMoreInfo component but I this might be better to update in the Link...

### CODE REVIEW

**Comment by willdurand** [comment]
> originally I updated just the AddonMoreInfo component but I this might be better to update in the Link...

Why not? I don't think we want to allow `href` with a `null` value.

**Comment by rebmullin** [comment]
> Why not? I don't think we want to allow href with a null value.

I guess I was thinking we might want to keep it this way to protect other cases where this might happen but I really wasn't sure. I'll revert this and recommit the update in the AddonMoreInfo component..

File: src/amo/components/AddonMoreInfo/index.js [hunk 1]
```diff
@@ -91,8 +91,25 @@ export class AddonMoreInfoBase extends React.Component<Props> {
       );
     }
 
-    const currentVersion = addon.current_version;
     const lastUpdated = addon.last_updated;
+    const currentVersion = addon.current_version;
```
Comment (willdurand) [inline src/amo/components/AddonMoreInfo/index.js]: I would have left this line where it was before.

File: src/amo/components/AddonMoreInfo/index.js [hunk 2]
```diff
@@ -91,8 +91,25 @@ export class AddonMoreInfoBase extends React.Component<Props> {
...
+    const { license } = currentVersion;
+    let licenseHtml = null;
```
Comment (willdurand) [inline src/amo/components/AddonMoreInfo/index.js]: You could probably name this variable `versionLicenseLink`, no?

File: tests/unit/amo/components/TestAddonMoreInfo.js [hunk 3]
```diff
@@ -185,6 +185,31 @@ describe(__filename, () => {
...
+    const addon = createInternalAddon({
+      ...fakeAddon,
+      current_version: {
+        ...fakeAddon.current_version,
+        license: { name: 'justText', url: null },
+      },
+    });
+    const root = render({ addon });
+    const link = root.find('.AddonMoreInfo-license-type');
+    expect(link.children()).toIncludeText('justText');
```
Comment (willdurand) [inline tests/unit/amo/components/TestAddonMoreInfo.js]: Where do you check that it is not a link?

File: tests/unit/amo/components/TestAddonMoreInfo.js [hunk 4]
```diff
@@ -185,6 +185,31 @@ describe(__filename, () => {
...
+    const addon = createInternalAddon({
+      ...fakeAddon,
+      current_version: {
+        ...fakeAddon.current_version,
+        license: null,
+      },
+    });
+    const root = render({ addon });
+    expect(root.find('.AddonMoreInfo-license')).toHaveLength(0);
+  });
```
Comment (willdurand) [inline tests/unit/amo/components/TestAddonMoreInfo.js]: I am surprised that this test case did not exist before.

File: src/amo/components/AddonMoreInfo/index.js [hunk 5]
```diff
@@ -91,8 +91,25 @@ export class AddonMoreInfoBase extends React.Component<Props> {
...
+        <Link
+          className="AddonMoreInfo-license-link"
+          href={license.url}
+          prependClientApp={false}
+          prependLang={false}
+        >
+          {license.name}
+        </Link>
+      ) : (
+        <span className="AddonMoreInfo-license-type">{license.name}</span>
```
Comment (willdurand) [inline src/amo/components/AddonMoreInfo/index.js]: Should it be `type` or `name`?

File: src/amo/components/AddonMoreInfo/index.js (hunk 2 above)
Comment (rebmullin) [reply src/amo/components/AddonMoreInfo/index.js]: well it's not always a link, right? so I thought this was more fitting...

File: src/amo/components/AddonMoreInfo/index.js (hunk 5 above)
Comment (rebmullin) [reply src/amo/components/AddonMoreInfo/index.js]: yeh I guess name : )

File: tests/unit/amo/components/TestAddonMoreInfo.js (hunk 3 above)
Comment (rebmullin) [reply tests/unit/amo/components/TestAddonMoreInfo.js]: Yep, i'll add another case : ) thanks

File: src/amo/components/AddonMoreInfo/index.js (hunk 2 above)
Comment (willdurand) [reply src/amo/components/AddonMoreInfo/index.js]: `versionLicenseLink` is the key used a few line below.

Comment (rebmullin) [reply src/amo/components/AddonMoreInfo/index.js]: Altho, I see that's what we are calling it in the function..

Comment (rebmullin) [reply src/amo/components/AddonMoreInfo/index.js]: okay I see..I'll just rename this

File: tests/unit/amo/components/TestAddonMoreInfo.js (hunk 4 above)
Comment (rebmullin) [reply tests/unit/amo/components/TestAddonMoreInfo.js]: yeh me too..

**Pull request merged**